src/util/temp-dir.js

```diff
@@ -63,7 +63,7 @@ export class TempDir {
    * been created.
    */
   create(): Promise<TempDir> {
-    const createTempDir = promisify(tmp.dir, {multiArgs: true});
+    const createTempDir = promisify(tmp.dir, tmp, {multiArgs: true});
```

Comment:
This seems to be the only user of `multiArgs`.

Since we only support Node >= 8, have you considered using the [built-in `util.promisify`](https://nodejs.org/api/util.html#util_util_promisify_original) and define a method using `util.promisify.custom` to have callback args?

Reply:
ah, that sounds good to me too, I didn't know they added this helper in nodejs 8.

I looked into that and update this PR accordingly.

For most of the usage `util.promisify` is already providing the behaviors we need, in some cases we need to explicitly bind the methods on their related object, but it isn't a big deal.

I repurposed the primisify module to provide a `multiArgsPromisedFn` helper, which is now used with promisify.custom to customize the `tmp.dir` behavior (and we can reuse it further if we met other cases that may need that), and so I also kept the unit test (to test the `multiArgsPromisedFn` helper, and also to make sure that the behaviors of the functions wrapped by `util.promisify` are going to be the expected one, e.g. when the original function throws).